var/spack/repos/builtin/packages/node-js/package.py

```diff
@@ -0,0 +1,106 @@
+#
+# For details, see https://github.com/llnl/spack
+# Please also see the LICENSE file for our notice and the LGPL.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License (as
+# published by the Free Software Foundation) version 2.1, February 1999.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the IMPLIED WARRANTY OF
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the terms and
+# conditions of the GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+##############################################################################
+from spack import *
+import sys
+import subprocess
+
+
+class NodeJs(Package):
+    """Node.js is a JavaScript runtime built on Chrome's V8 JavaScript
+    engine."""
+
+    homepage = "https://nodejs.org/"
+    url      = "https://nodejs.org/download/release/v6.3.0/node-v6.3.0.tar.gz"
+
+    version('7.1.0', '1db5df2cb025f9c70e83d9cf21c4266a')
+    version('6.3.0', '8c14e5c89d66d4d060c91b3ba15dfd31')
+    version('6.2.2', '1120e8bf191fdaee42206d031935210d')
+
+    # variant('bash-completion', default=False, description='Build with bash-completion support for npm')  # NOQA: ignore=E501
+    variant('debug',           default=False, description='Include debugger support')
+    variant('doc',             default=False, description='Compile with documentation')
+    variant('icu4c',           default=False, description='Build with support for all locales instead of just English')
+    variant('openssl',         default=True,  description='Build with Spacks OpenSSL instead of the bundled version')
+    variant('zlib',            default=True,  description='Build with Spacks zlib instead of the bundled version')
+
+    # depends_on('libtool',         type='build') # if sys.platform != 'darwin'
+    depends_on('pkg-config',      type='build')
+    depends_on('python@2.7:',     type='build')
+    # depends_on('bash-completion', when="+bash-completion")
+    depends_on('icu4c',           when='+icu4c')
+    depends_on('openssl',         when='+openssl')
+
+    def install(self, spec, prefix):
+        options = []
+        options.extend(['--prefix={0}'.format(prefix)])
+
+        # Note: npm is updated more regularly than node.js, so we build the
+        #       package instead of using the bundled version
+        options.extend(['--without-npm'])
+
+        # On OSX, the system libtool must be used
+        # So, we ensure that this is the case by...
+        if sys.platform == 'darwin':
+            result_which = subprocess.check_output(["which", "libtool"])
+            result_whereis = subprocess.check_output(["whereis", "libtool"])
```

Comment:
See: http://stackoverflow.com/questions/4814970/subprocess-check-output-doesnt-seem-to-exist-python-2-6-5